components/needs_team/needs_team.tsx

```diff
@@ -224,6 +227,13 @@ export default class NeedsTeam extends React.Component<Props, State> {
         this.props.actions.loadStatusesForChannelAndSidebar();
         this.props.actions.loadProfilesForDirect();
 
+        this.props.actions.getAllGroupsAssociatedToChannelsInTeam(team.id, true);
+        if (team.group_constrained) {
+            this.props.actions.getAllGroupsAssociatedToTeam(team.id, true);
+        } else {
+            this.props.actions.getGroups(true);
```

Comment:
As discussed, I think this call to `getGroups` should be invoked once rather than each time the user changes teams.

Reply:
agree